src/connector/src/connector_common/mqtt_common.rs

```diff
@@ -111,6 +116,14 @@ impl MqttCommon {
 
         options.set_clean_start(self.clean_start);
 
+        if let Some(max_packet_size) = self.max_packet_size {
+            options.set_connect_properties({
+                let mut props = ConnectProperties::new();
+                props.max_packet_size = Some(max_packet_size);
+                props
+            });
+        }
```

Comment:
This `if let` seems superfluous, couldn't it just be:

```rust
options.set_connect_properties({
    let mut props = ConnectProperties::new();
    props.max_packet_size = self.max_packet_size;
    props
});
```

Reply:
It is not entirely the same. In my implementation does not add an empty ConnectProperties when there is no max_packet_size given.

Reply:
Let's move the `ConnectProperties` to the upper level in case we add more properties.

```
let mut props = ConnectProperties::new();
props.max_packet_size = self.max_packet_size;
options.set_connect_properties(props);
```